src/core/sagas/utils.js

```diff
@@ -1,19 +1,17 @@
-/* @flow */
 import { ErrorHandler } from 'core/errorHandler';
-import log from 'core/logger';
-import type { ApiStateType } from 'core/reducers/api';
+import defaultLog from 'core/logger';
 
-export function createErrorHandler(id: string): typeof ErrorHandler {
+
+export function createErrorHandler(id, { log = defaultLog } = {}) {
   return new ErrorHandler({
     id,
     // Make sure the dispatch() method can't be used. A saga will yield
     // put(action) instead so this shouldn't cause a problem.
-    dispatch: () => log.error(
-      'ErrorHandler cannot dispatch from a saga'),
+    dispatch: () => log.error('ErrorHandler cannot dispatch from a saga'),
   });
 }
 
-// Convenience function to extract API info.
-export function getApi(state: {| api: ApiStateType |}): ApiStateType {
-  return state.api;
+// Convenience function to extract state info.
+export function getState(state) {
```

Comment:
I find the syntax for objects with default values (eg `{ log = defaultLog }: {| log: typeof defaultLog |} = {}`) to be totally hostile and it's really frustrating to me. But I think I've made the best case I can against flow already which is that I find the syntax to be actively distracting.

I think `state` is largely understood to be our state object, the result of `store.getState()` from our redux `store` object. That's not intuitive if you haven't used a react/redux app (or maybe even haven't used our app), but I think flow's annotations are even less intuitive than the `state` object in this function.

When we're often declaring the types in a separate file I still need to check somewhere else to see exactly what `ApiStateType` is anyway. I've yet to hit a bug that flow would have caught. That's perhaps a bit chicken and egg because we lack any substantial flow coverage, but to me the syntax fights with me more than the bugs it would catch were we to have coverage.